blockchain/blockdao.go

```diff
@@ -606,6 +705,135 @@ func (dao *blockDAO) deleteTipBlock() error {
 	return dao.kvstore.Commit(batch)
 }
 
+// getDBFromHash returns db of this block stored
+func (dao *blockDAO) getDBFromHash(h hash.Hash256) (db.KVStore, uint64, error) {
+	height, err := dao.getBlockHeight(h)
+	if err != nil {
+		return nil, 0, err
+	}
+	return dao.getDBFromHeight(height)
+}
+
+func (dao *blockDAO) getTopDB(blkHeight uint64) (kvstore db.KVStore, index uint64, err error) {
+	if dao.cfg.SplitDBSizeMB == 0 {
+		return dao.kvstore, 0, nil
+	}
+	if blkHeight <= dao.cfg.SplitDBHeight {
+		return dao.kvstore, 0, nil
+	}
+	topIndex := dao.topIndex.Load().(uint64)
+	file, dir := getFileNameAndDir(dao.cfg.DbPath)
+	if err != nil {
+		return
+	}
+	longFileName := dir + "/" + file + fmt.Sprintf("-%08d", topIndex) + ".db"
+	dat, err := os.Stat(longFileName)
+	if err != nil && os.IsNotExist(err) {
+		// db file is not exist,this will create
+		return dao.openDB(topIndex)
+	}
+	// other errors except file is not exist
+	if err != nil {
+		return
+	}
+	// dao.cfg.SplitDBSize need convert to M
+	if uint64(dat.Size()) > dao.cfg.SplitDBSize() {
```

Comment:
You can choose to only create a new db file when you want to insert block, but not receipt to make sure block and receipt are always of the same height are the same db file. This is going to prevent your passing kvstore reference around

Reply:
BTW, should this check be done earlier? If we want to create a new db file, we wouldn't want to open the previous one (L733)?